### PR TITLE
Docker deploy

### PR DESCRIPTION
**Pull request by vickumar1981**
Allows local development using docker and docker-compose

## Additions

- Adds docker configuration: Dockerfiles for containers and a docker-compose.yml

## Removals

-

## Changes

-

## Testing

- brew install docker docker-compose and run docker-compose up. Should have cf-gov website with optional apps installed and running locally.

## Review

- @rosskarchner @willbarton @kurtw @chosak

## Notes

- Currently, the container is not compiling all the front end assets for all the apps. I believe this is an issue with the specific versions of node and npm inside the container.
- Added a DOCKER.md file to describe the process.

## Todos

- Add PDF reactor to the docker container
- Update npm and node versions so frontend.sh scripts work for all apps

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by vickumar1981** [comment]
Please see https://github.com/cfpb/cfgov-refresh/pull/2224.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ENV MYSQL_NAME "v1"
+ENV MYSQL_USER "v1"
+ENV MYSQL_PW "password"
+ENV MYSQL_HOST "mysql"
+ENV ES_HOST "search"
+
+ENV STAGING_HOSTNAME "content.localhost"
+ENV DJANGO_STATIC_ROOT "/code/cfgov-refresh/cfgov/static_built"
+ENV WAGTAIL_ADMIN_PW "admin"
+ENV ADMIN_EMAILS "emmanuel.apau@excella.com;kaveiii@gmail.com"
```
Comment (rosskarchner) [inline Dockerfile]: He doesn't work here anymore?

Comment (vickumar1981) [inline Dockerfile]: @rosskarchner   nice catch.  Fixed.

File: cfgov/cfgov/redirects.py [hunk 2]
```diff
@@ -0,0 +1,661 @@
+from django.views.generic.base import TemplateView, RedirectView
+from django.conf.urls import include, url
+
+def rewrite_url(url1, url2, **kwargs):
+    return url(url1, RedirectView.as_view(url=url2, **kwargs))
+
+urlredirects =  [
+
+]
+# # these used to live in httpd.conf
```
Comment (vccabral) [inline cfgov/cfgov/redirects.py]: this is garbage.

File: cfgov/cfgov/wsgi.py [hunk 3]
```diff
@@ -10,7 +10,9 @@
 import os
 
 from django.core.wsgi import get_wsgi_application
+#from dj_static import Cling
```
Comment (vccabral) [inline cfgov/cfgov/wsgi.py]: fix me

File: docker/local.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+cp -R -n /apps/ .
```
Comment (vccabral) [inline docker/local.sh]: should we allow the devs to decide when to run the server?

**Comment by rosskarchner** [comment]
Thanks for the work you've done on this @vickumar1981 @vccabral . I think for the moment we're going to close this PR -- it's got a lot of ideas in it, and we want to discuss them before accepting them.

**Pull request closed without merging**